### PR TITLE
Issue 17850 reserved vars

### PR DESCRIPTION
**Pull request by wezell**

### CODE REVIEW

**Comment by nollymar** [comment]
Please, add an IT

**Comment by jcastro-dotcms** [comment]
@wezell do these changes handle the problem for customers with existing data having a now-invalid variable name?

**Comment by wezell** [comment]
@jcastro-dotcms  - the content and content type would already be broken and unable be indexed - and as for updating var names, there is really no way to safely do this as the var names are probably used in queries in widgets, .vtls etc...

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java [hunk 1]
```diff
@@ -316,7 +367,9 @@ private ContentType dbSaveUpdate(final ContentType saveType) throws DotDataExcep
     }
 
     ContentType retType = builder.build();
-
+    if(reservedContentTypeVars.contains(retType.variable().toLowerCase())){
+        throw new IllegalArgumentException("Invalid content type variable - reserved var name: " + retType.variable().toLowerCase());
```
Comment (oarrietadotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: sorry to bother, we could use a Logger here, for tracking purposes.

**Review by nollymar — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 2]
```diff
@@ -631,40 +631,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
-    @DataProvider
-    public static List<Object> dataProviderEEContentTypes() throws Exception {
-        // data provider needs stuff to get initialized because of API access
-        IntegrationTestInitService.getInstance().init();
+    @Test
+    public void testGetSchema_GivenNoEELicense_EnterpriseTypesShouldNotBeAvailableInSchema() throws Exception{
 
         // filter only Enterprise content types
-        final List<ContentType> eeTypes = APILocator
+        List<ContentType> eeTypes = APILocator
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4826945)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 3]
```diff
@@ -631,40 +631,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
-                    testCase._2);
-
-            runNoLicense(() -> {
-                final GraphQLSchema schema = APILocator.getGraphqlAPI().getSchema();
-                assertNull(schema.getType(testCase._1));
-            });
-        } finally {
-            if(customType!=null) {
-                APILocator.getContentTypeAPI(APILocator.systemUser()).delete(customType);
+        List<Tuple2<String, BaseContentType>> eeTypesList = eeTypes.stream().map((type)->
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4826945)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 4]
```diff
@@ -631,40 +631,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
 
     }
 
-    @DataProvider
-    public static List<Object> dataProviderEEContentTypes() throws Exception {
-        // data provider needs stuff to get initialized because of API access
-        IntegrationTestInitService.getInstance().init();
+    @Test
+    public void testGetSchema_GivenNoEELicense_EnterpriseTypesShouldNotBeAvailableInSchema() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4826945)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 5]
```diff
@@ -631,40 +631,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
-                final GraphQLSchema schema = APILocator.getGraphqlAPI().getSchema();
-                assertNull(schema.getType(testCase._1));
-            });
-        } finally {
-            if(customType!=null) {
-                APILocator.getContentTypeAPI(APILocator.systemUser()).delete(customType);
+        List<Tuple2<String, BaseContentType>> eeTypesList = eeTypes.stream().map((type)->
+                        new Tuple2<>("my"+type.variable(), type.baseType())).collect(Collectors.toList());
+
+        for (Tuple2<String, BaseContentType> testCase : eeTypesList) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4826945)

**Review by jgambarios — APPROVED** [review]

**Pull request merged**